pkg/api/register.go

```diff
@@ -118,52 +118,67 @@ func AddToScheme(scheme *runtime.Scheme) {
 func (obj *Pod) GetObjectMeta() meta.Object                                  { return &obj.ObjectMeta }
 func (obj *Pod) GetObjectKind() unversioned.ObjectKind                       { return &obj.TypeMeta }
 func (obj *PodList) GetObjectKind() unversioned.ObjectKind                   { return &obj.TypeMeta }
+func (obj *PodList) GetListMeta() meta.List                                  { return &obj.ListMeta }
```

Comment:
Can you test with it to verify it works?  The API test I mentioned would
definitely cover us (verify a set/get on all types and that they implement
the interface).

On Tue, Jun 7, 2016 at 9:17 PM, Andy Goldstein <notifications@github.com>
wrote:

> In pkg/api/register.go
> <https://github.com/kubernetes/kubernetes/pull/26964#discussion_r66180119>
> :
>
> > @@ -118,52 +118,67 @@ func AddToScheme(scheme *runtime.Scheme) {
> >  func (obj *Pod) GetObjectMeta() meta.Object                                  { return &obj.ObjectMeta }
> >  func (obj *Pod) GetObjectKind() unversioned.ObjectKind                       { return &obj.TypeMeta }
> >  func (obj *PodList) GetObjectKind() unversioned.ObjectKind                   { return &obj.TypeMeta }
> > +func (obj *PodList) GetListMeta() meta.List                                  { return &obj.ListMeta }
>
> I will happily change this to rip out GetObjectMeta and my newly-added
> GetListMeta if there's a better way
>
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub
> <https://github.com/kubernetes/kubernetes/pull/26964/files/6b0110a098af230d6fd5c3ad2b171ab4730c50dc#r66180119>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/ABG_p8tVcb8HfFonWHg_ulPh2lHrt71Jks5qJhhFgaJpZM4Iv_PB>
> .
>